src/internal/native/array.ts

```diff
@@ -4,6 +4,23 @@
 import { ImageMagick } from '../../image-magick';
 import { quantumArray } from '@dlemstra/magick-native/magick';
 
+/** @internal */
+export function _withArray<TReturnType>(array: Uint8Array | Uint8ClampedArray, func: (instance: number) => TReturnType): TReturnType {
+    const length = array.length;
+    if (length === 0)
+        return func(0);
```

Comment:
Thanks for adding this check, that's something I missed. But I think we should throw an error here instead.